idx/memory/memory.go

```diff
@@ -119,15 +122,19 @@ func (t *TagIndex) addTagId(name, value string, id schema.MKey) {
 	ti[name][value][id] = struct{}{}
 }
 
-func (t *TagIndex) delTagId(name, value string, id schema.MKey) {
+func (t *TagIndex) delTagId(name, value string, id schema.MKey, m *MemoryIdx) {
 	ti := *t
 
 	delete(ti[name][value], id)
 
 	if len(ti[name][value]) == 0 {
 		delete(ti[name], value)
+		vPtr, _ := m.objIntern.GetNoRefCnt([]byte(value))
+		m.internReleasePtr(vPtr)
```

Comment:
would it make sense to add some `DeleteByValue()` method to the object store? then the call to `internReleasePtr` could be saved, and also its check whether the pointer is `0`

Reply:
Yes, I guess that's what I'm doing anyway, just wasting function calls in between. I think that would work well with string optimized methods too. I'll implement those in the interning layer and change it up here.